src/clarity_vm/special.rs

```diff
@@ -566,6 +593,7 @@ fn synthesize_pox_2_event_info(
         "stack-aggregation-commit"
         | "stack-aggregation-commit-indexed"
         | "stack-aggregation-increase" => Some(create_event_info_aggregation_code(function_name)),
+        "delegate-stx" => Some(create_event_info_stack_or_delegate_code(sender, function_name, args)),
```

Comment:
Should this be moved to the first match condition which already calls `create_event_info_stack_or_delegate_code`?